openassessment/xblock/studio_mixin.py

```diff
@@ -152,7 +156,7 @@ def editor_context(self):
             'file_upload_response': self.file_upload_response if self.file_upload_response else '',
             'necessity_options': self.NECESSITY_OPTIONS,
             'file_upload_type': self.file_upload_type,
-            'white_listed_file_types': self.white_listed_file_types_string,
```

Comment:
question. Here we were referencing `self.white_listed_file_types_string` and  in `get_submission_path_and_context()` we were referencing `self.white_listed_file_types` ... do we just store the same value on the block twice? is one a derived @property? Do we update them together? In `update_editor_context()` it only looks like we're saving `self.white_listed_file_types_string`?

Reply:
Annoyingly, we store both the `white_listed_file_types` (an array of file extensions) and `white_listed_file_types_string` (a comma delimited list of extensions). The array is used for extension validation in `upload_url` and the string is used in the `editor_context` and serialization/deserialization from XML.

Reply:
The string is the actual persisted value, but the string setter also parses the string and sets the array variable for use around the ORA.